__tests__/fetchers.js

```diff
@@ -433,3 +433,78 @@ test('TarballFetcher.fetch throws on truncated tar header', async () => {
     new RegExp(reporter.lang('errorExtractingTarball', '.*', '.*broken-tar-header.tgz')),
   );
 });
+
+test('TarballFetcher.fetch retries on a truncated response', async () => {
+  const dir = await mkdir('tarball-fetcher');
+  const config = await Config.create({});
+
+  const workingUrl = 'https://registry.npmjs.org/lodash.isempty/-/lodash.isempty-4.4.0.tgz';
+  // This one returns a truncated response - body is shorter than the Content-Length header.
+  const brokenUrl = 'https://registry.npmjs.org/lodash.isempty/-/lodash.isempty-4.4.0-broken.tgz';
+
+  const mockRequest = (require: any).requireActual('./__mocks__/request');
+
+  let retryCount = 2;
+
+  // Shorter retry delay for tests
+  config.requestManager.retryDelay = 10;
+  config.requestManager._requestModule = options => {
+    retryCount -= 1;
+    if (retryCount >= 0) {
```

Comment:
Why not `(retryCount > 0)` here and `retryCount = 3` above and `retryCount <= 0` below? Looks clearer to me.

Reply:
Since it's two retries (3 requests total). But can improve the wording and logic a bit to make it clearer.